PrebidMobile/PrebidMobileCore/src/main/java/org/prebid/mobile/core/CacheManager.java

```diff
@@ -0,0 +1,84 @@
+package org.prebid.mobile.core;
+
+import android.content.Context;
+import android.text.TextUtils;
+import android.webkit.WebView;
+
+import java.util.HashMap;
+import java.util.UUID;
+
+public class CacheManager {
+    // This is the class that manages three cache instances, one for DFP, one for MoPub and one for SDK rendered ad
+    private WebView dfpWebCache;
+    private WebView mopubWebCache;
+    private HashMap<String, String> sdkCache;
+    private static CacheManager cache;
+
+    private CacheManager(Context context) {
+        setupDFPCache(context);
+        setupMoPubCache(context);
+        setupSDKCache();
+    }
+
+    static void init(Context context) {
+        if (cache == null) {
+            cache = new CacheManager(context);
+        }
+    }
+
+    public static CacheManager getCacheManager() {
+        return cache;
+    }
+
+    public String saveCache(String bid, String format) {
+        if (TextUtils.isEmpty(bid)) {
+            return null;
+        }
+        String cacheId = UUID.randomUUID().toString();
```

Comment:
So for the iOS solution, i'm considering setting the `cacheId` on the server still, even if we don't cache the bid (ie just generating a UUID on server instead of here).

That way we don't have to update the keys on the client side and it will work seamlessly for MoPub iOS solution where we still need to cache.

Thoughts??

Reply:
I was thinking about having timestamp in the cache id while working on the pbm.js since we need to have a mechanism to expire the bids after a while. If generating the cache id on the server then this won't be able to work.
Thoughts?

Reply:
Synced offline. Generating the cache id client side will work.